### DISCUSSION TITLE
zoom and pan

### DISCUSSION THREAD

**Discussion by francescocerasuolo** [observablehq/plot#763] [Ideas]
Is it possible to allow users to interact with a scatter plot, by performing zoom and pan around the plot?

**Comment by Fil** [comment]
Great question! Interactions are on the roadmap, but for now you have to experiment. Here's a few examples that might help get you started: https://observablehq.com/@observablehq/plot-zoom .

**Reply by wiless** [reply]
I tried to play around zoom/pan using the d3.zoom https://observablehq.com/@wiless/playing-with-zoom-pan-with-svg-and-plots#myplot , I could manage axis scaling by redrawing using d3.axis , but not sure how to handle a) the marks getting outside the AXIS boundary b) the vanishing grid lines once you zoom-in and zoom-out. 

I am not a expert at SVG or D3js users.. Please advise .. @Fil  .

**Reply by Fil** [reply]
thanks, that's very helpful! here's a version that fixes both the clipping and the grid lines issues:
https://observablehq.com/@fil/playing-with-zoom-pan-with-svg-and-plots-suggestion

You'll see that they'll be much easier if we could have the grid lines as options of d3-axis (rather than add-ons in Plot), and that adding *selection*.wrap() to d3-selection would also help.

**Reply by romanzenka** [reply]
I am very confused by this library. It is built on top of d3, and it came from Observable.. but when I read the source code, I cannot spot any place where reactive programming or d3-style updates are actually enabled. What am I missing? How can this allow smooth zooms unless one "takes apart" the resulting SVG, and zooms it "by hand", as your examples show?

My impression so far is that this is the "forward pass" - go from data to picture. What I do not see is "go from data that changes in time to a picture that changes in time (somehow optimally)". Is that in scope of this project at all?

**Reply by enjalot** [reply]
Updating data via transitions and interactions are not yet supported, but will certainly be worked on. The approach of Plot has been to build a solid foundation and solicit feedback as it develops.
Examples and experiments are very welcome, as Fil pointed out it is possible to use d3 to extend Plot to attempt these kinds of things. The more attempts the clearer the final abstraction becomes.
